blockchain/provider.go

```diff
@@ -119,26 +119,26 @@ func (p *Provider) GetAppCoinFrozenStatus(
 }
 
 func (p *Provider) GetAppCoinBalanceAndFrozenStatus(
-	callOpts *bind.CallOpts, coin, address common.Address) (int64, int64, error) {
+	callOpts *bind.CallOpts, coin, address common.Address) (*big.Int, int64, error) {
 	balance, err := p.GetAppCoinBalance(callOpts, coin, address)
 	if err != nil {
-		return 0, 0, errors.WithMessage(err, "failed to get balance")
+		return nil, 0, errors.WithMessage(err, "failed to get balance")
 	}
 
 	frozen, err := p.GetAppCoinFrozenStatus(callOpts, coin, address)
 	if err != nil {
-		return 0, 0, errors.WithMessage(err, "failed to get frozen status")
+		return nil, 0, errors.WithMessage(err, "failed to get frozen status")
 	}
 
 	return balance, frozen, err
 }
 
 // GetAppCoinBalance gets APP coin balance for specific address.
 func (p *Provider) GetAppCoinBalance(
-	callOpts *bind.CallOpts, coin, address common.Address) (int64, error) {
+	callOpts *bind.CallOpts, coin, address common.Address) (*big.Int, error) {
 	appCoinContract, err := p.GetAppCoinContract(coin)
 	if err != nil {
-		return 0, errors.WithMessage(err, "failed to get APP coin contract")
+		return nil, errors.WithMessage(err, "failed to get APP coin contract")
 	}
 
 	balance, err := appCoinContract.BalanceOf(callOpts, address)
```

Comment:
need support airdrop, should call `balanceOfWithAirdrop`

Reply:
ABI will be regenerated late.